instrumentation/instaredigo/instaredigo.go

```diff
@@ -0,0 +1,303 @@
+}
+
+// Send writes the command to the client's output buffer and collect the
+// instrumentation details.Inorder to capture the correlated span information,
+// create a context from the parent span and pass it as an argument along with
+// the other arguments. The Send API will retrieve the span information from the
+// context and record the correlated span information.
+func (c *instaRedigoConn) Send(commandName string, args ...interface{}) (err error) {
+	var cmdArgs []interface{}
+	// Separating the parent context from the arguments
+	ctx := context.Background()
+	for _, a := range args {
+		if tmp, ok := a.(context.Context); ok {
+			ctx = tmp
+		} else {
+			cmdArgs = append(cmdArgs, a)
+		}
+	}
+
+	// Handling the command if there exists a previous span
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.prevSpan != nil {
+
+		err = c.Conn.Send(commandName, cmdArgs...)
+		if err != nil {
+			c.prevSpan.span.SetTag("redis.error", err.Error())
+			c.prevSpan.span.LogFields(otlog.Object("error", err.Error()))
+		}
+
+		if !isTransactionEnding(commandName) {
+			c.prevSpan.batchCmds = append(c.prevSpan.batchCmds, commandName)
+			c.prevSpan.span.SetTag("redis.subCommands", c.prevSpan.batchCmds)
+		} else {
+			c.prevSpan.span.Finish()
+			c.prevSpan = nil
+		}
+
+	} else {
+
+		// Fetching the tracer
+		tracer := c.sensor.Tracer()
+		opts := []ot.StartSpanOption{
+			ot.Tags{
+				"redis.connection": c.address,
+			},
+		}
+
+		// Fetching the span and starting it
+		if ps, ok := instana.SpanFromContext(ctx); ok {
+			tracer = ps.Tracer()
+			opts = append(opts, ot.ChildOf(ps.Context()))
+		}
+		span := tracer.StartSpan("redis", opts...)
+
+		// Checking for multi command transaction
+		if isTransactionBeginning(commandName) {
+			c.prevSpan = &prevSpan{span, nil}
+		} else {
+			defer span.Finish()
+		}
+
+		// Setting the span tags and executing the command
+		span.SetTag("redis.command", commandName)
+		err = c.Conn.Send(commandName, cmdArgs...)
+		if err != nil {
+			span.SetTag("redis.error", err.Error())
+			span.LogFields(otlog.Object("error", err.Error()))
+		}
+	}
+
+	return err
+}
+
+func isTransactionEnding(commandName string) bool {
+	return strings.ToUpper(commandName) == "EXEC" || strings.ToUpper(commandName) == "DISCARD"
+}
+
+func isTransactionBeginning(commandName string) bool {
+	return strings.ToUpper(commandName) == "MULTI"
+}
+
+func (c *instaRedigoConn) genericDo(ctx context.Context, timeout time.Duration,
+	commandName string, args ...interface{}) (interface{}, error) {
+
+	if cwc, ok := c.Conn.(redis.ConnWithContext); ok {
+
+		return cwc.DoContext(ctx, commandName, args...)
+	} else if cwt, ok := c.Conn.(redis.ConnWithTimeout); ok {
+
+		return cwt.DoWithTimeout(timeout, commandName, args...)
+	} else {
+
+		return c.Conn.Do(commandName, args...)
+	}
+}
+
+func (c *instaRedigoConn) handleMultiTransaction(commandName string,
+	ctx context.Context, timeout time.Duration,
```

Comment:
`timeout` is not used in the function

Reply:
timeout is needed instead of default value. Updated it